File: deaf_grandma.js
function deafGrandma() {
    let count = 0;
    
    gabbyGrams(window.prompt("HEY KID!"))
    function gabbyGrams(initPrompt) {
        let resp = initPrompt;
        if (resp === ""){
                gabbyGrams(window.prompt("WHAT?!"));
        } else if (resp === 'GOODBYE!') {
            if (count === 1){
                return count +=1;
            } else {
                count += 1;
                gabbyGrams(window.prompt("LEAVING SO SOON?"));
            }
        } else if (resp.split('').map(elem => elem.replace(/[^a-zA-Z]/, '')).join('').split('').every(elem => elem.match(/[A-Z]/))){
            gabbyGrams(window.prompt('NO, NOT SINCE 1946!'));
        } else {
            gabbyGrams(window.prompt('SPEAK UP, KID!'));
        }
    }
    return alert("LATER, SKATER!")
}

deafGrandma();

File: deaf_grandma.py
def deaf_grandma():
    bye_count = 0
    prompt = input("HEY KID!")
    while bye_count != 2:
        if prompt == "":
            prompt = input("WHAT?!")
        elif prompt == "GOODBYE!":
            if bye_count == 0:
                prompt = input("LEAVING SO SOON?")
                bye_count += 1
            elif bye_count == 1:
                bye_count += 1
                print("LATER, SKATER!")
        elif prompt.isupper():
            prompt = input("NO, NOT SINCE 1946!")
        else:
            prompt = input("SPEAK UP, KID!")
deaf_grandma()
